lib/current-env.js

```diff
@@ -1,9 +1,6 @@
 const process = require('node:process')
 const nodeOs = require('node:os')
-
-function isMusl (file) {
-  return file.includes('libc.musl-') || file.includes('ld-musl-')
-}
+const { familySync } = require('detect-libc')
```

Comment:
Only in very rare cases, if ever (because it's only on linux as a fallback to reading the ldd file and the getReport)
https://github.com/lovell/detect-libc/blame/2642d9612fadf14f3eeb5cd56eb3a63b254c7693/lib/detect-libc.js#L144-L154

But if it's a problem we can just take the read from ldd file logic which will solve 99% of problems

Reply:
I think the ldd file logic is likely the way to go.

Reply:
This PR already has some of it, albeit by using requiremock.  We use tap so we'd just want to use tap.mock.

I'd say `current-env` is already sufficiently isolated to be ready for mocking.

Perhaps mocking isn't even the right option here.  The require cache is the problem.  cleaning it as a `t.afterEach` may be sufficient.

Reply:
```
delete require.cache[require.resolve(module)]
```